README.md

```diff
@@ -0,0 +1,108 @@
+# cargo-vet
+

```

Comment:
ah, sample-criteria helps a bit. it's... interesting that it seems to conflate "methodology" and "claims"? Like "reviewed" feels like a very different *kind* of statement to "sound". hmm...

also is there at all a subsetting relationship between "sound" and "secure"? It feels like there should be but I'm genuinely not sure in which direction.

Reply:
oh wow actually your changes seem way more wide-sweeping than i expected, ok I really just need to re-read this completely

Reply:
> default-criteria: "The value may be either a string or an array of strings."
> 
> I assume this implies criteria in audited.toml is also like this?

Yes.

> Are you picturing something like "reviewed, fuzzed, tested"?

I was more imagining a situation wherein Firefox audits for `secure`, and imports audits from another project that does a lot of audits for `sound`. In that case, there could be value in splitting `secure` into two sub-criteria, so that we could skip soundness audits for some crates and just focus on ambient capabilities.

It's quite possible that this is too complex to be useful in practice, so we could also introduce it later as needed, but allowing criteria to be an array struck me as a nice generalization which didn't have any obvious downsides.

> ah, sample-criteria helps a bit. it's... interesting that it seems to conflate "methodology" and "claims"? Like "reviewed" feels like a very different _kind_ of statement to "sound". hmm...

That's a fair point. I had originally named them "correct" and "crypto-correct", but then I got worried that it would feel like an unreasonably high bar (zero bugs in this code), whereas the practical policy would be more about applying first-party-level scrutiny. I'm open to various outcomes here, though since it doesn't affect the implementation we can probably punt on it for now.

> also is there at all a subsetting relationship between "sound" and "secure"? It feels like there should be but I'm genuinely not sure in which direction.

As written, "sound" is almost a strict subset of "secure", except for the matter of degree (i.e. "secure" makes some allowances for discretion, whereas "sound" doesn't). I think we want the discretion in "secure" for Firefox's use-case, but I thought I'd get a lot of grief from soundness purists by introducing similar discretion in "sound".